### PR TITLE
fix: return error codes

### PR DESCRIPTION
**Pull request by xiduzo**
Return error codes, instead of static text, to the frontend and have the frontend do the translations accordingly

Solves PZ-4584

### CODE REVIEW

**Review by infonl-marcel — APPROVED** [review]
Great to move into this solution direction! 👍🏽

File: src/main/app/src/assets/i18n/en.json [hunk 1]
```diff
@@ -759,6 +759,8 @@
   "msg.error.besluit.publication.date.missing": "Missing decision publication date",
   "msg.error.besluit.response.date.missing": "Missing decision response date",
   "msg.error.besluit.response.date.invalid": "Response date for decision should not be before publication date",
+  "msg.error.case.has.open.subcases": "This case has open subcases and can not be closed",
```
Comment (edgarvonk) [inline src/main/app/src/assets/i18n/en.json:762]:
```suggestion
  "msg.error.case.has.open.subcases": "This case has open subcases and cannot be closed",
```

https://www.merriam-webster.com/grammar/cannot-vs-can-not-is-there-a-difference ;-)

File: src/main/app/src/assets/i18n/en.json [hunk 2]
```diff
@@ -759,6 +759,8 @@
...
+  "msg.error.case.has.locked.information.objects": "this case has locked information objects and can not be closed",
```
Comment (edgarvonk) [inline src/main/app/src/assets/i18n/en.json:763]:
```suggestion
  "msg.error.case.has.locked.information.objects": "This case has locked documents and cannot be closed",
```

As far as I know we don't use the term 'information objects' in the UI.

**Review by edgarvonk — CHANGES_REQUESTED** [review]

File: src/main/app/src/assets/i18n/en.json (hunk 1 above)
Comment (edgarvonk) [reply src/main/app/src/assets/i18n/en.json:762]: And: close the sentence with a `.` or not?

**Review by edgarvonk — CHANGES_REQUESTED** [review]

File: src/main/app/src/assets/i18n/nl.json [hunk 3]
```diff
@@ -759,6 +759,8 @@
   "msg.error.besluit.publication.date.missing": "Ontbrekende publicatiedatum van besluit",
   "msg.error.besluit.response.date.missing": "Ontbrekende uiterlijke reactiedatum van besluit",
   "msg.error.besluit.response.date.invalid": "De uiterlijke reactiedatum voor het besluit mag niet vóór de publicatiedatum liggen",
+  "msg.error.case.has.open.subcases": "Deze hoofdzaak heeft open deelzaken en kan niet afgesloten worden",
```
[the remark this answers is not in the document]
Comment (bas-info-nl) [reply src/main/app/src/assets/i18n/nl.json:762]: None of the messages have a `.` at the end. Those are added when building the onscreen message I believe, can you confirm @xiduzo ?

**Review by bas-info-nl — APPROVED** [review]

File: src/main/app/src/assets/i18n/nl.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (xiduzo) [reply src/main/app/src/assets/i18n/nl.json:762]: I do not see anything for this in the code-base. But to keep them consitent I would either add an `.` to all of the errors or tot none.

[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/app/src/assets/i18n/nl.json:762]: >None of the messages have a . at the end. 

That's not true! Most of the errors message do (scroll up). Better to be consistent here I feel.

**Review by edgarvonk — APPROVED** [review]

**Pull request merged**